pkg/cmd/kafka/create/data.go

```diff
@@ -0,0 +1,160 @@
+package create
+
+import (
+	"github.com/redhat-developer/app-services-cli/pkg/shared/accountmgmtutil"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/connection"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/factory"
+	kafkamgmtclient "github.com/redhat-developer/app-services-sdk-go/kafkamgmt/apiv1/client"
+)
+
+// Types we use on backend to map AMS Quotas
+type CloudProviderId = string
+
+// Additional types that are used in backend to match
+// AMS Quota instance types
+const (
+	// Matches QuotaTrialType = trial
+	DeveloperType CloudProviderId = "developer"
+
+	// Matches QuotaStandardType = standard
+	StandardType CloudProviderId = "standard"
+
+	// Deprecated by DeveloperType
+	TrialType CloudProviderId = "eval"
+)
+
+// mapAmsTypeToBackendType - Cloud providers API is not using AMS types but some other values (CloudProviderValues)
+func mapAmsTypeToBackendType(amsType *accountmgmtutil.QuotaSpec) CloudProviderId {
+	switch amsType.Name {
+	case accountmgmtutil.QuotaStandardType:
+		return StandardType
+	case accountmgmtutil.QuotaTrialType:
+		return DeveloperType
+	default:
+		return DeveloperType
+	}
+}
+
+// return list of the valid instance sizes for the specifed region and ams instance types
+func GetValidKafkaSizes(f *factory.Factory,
+	providerID string, regionId string, amsType *accountmgmtutil.QuotaSpec) ([]string, error) {
+
+	conn, err := f.Connection(connection.DefaultConfigSkipMasAuth)
+	if err != nil {
+		return nil, err
+	}
+
+	validSizes := []string{}
+
+	instanceTypes, _, err := conn.API().
+		KafkaMgmt().
+		GetInstanceTypesByCloudProviderAndRegion(f.Context, providerID, regionId).
+		Execute()
+	if err != nil {
+		return nil, err
+	}
+
+	var desiredInstanceType string
+	if amsType != nil {
+		desiredInstanceType = mapAmsTypeToBackendType(amsType)
+	} else {
+		// If we do not know instance type from AMS we should assume standard instance type
```

Comment:
This logic seems dangerous to me. The kas-fleet-manager api should somehow provide the default value.

Reply:
In other words, I think the else here should be an error condition.

Reply:
I have removed need to have else :)